mlrun/api/db/sqldb/db.py

```diff
@@ -665,6 +674,204 @@ def delete_project(self, session, name: str):
     def list_projects(self, session, owner=None):
         return self._query(session, Project, owner=owner)
 
+    def get_feature_set(
+        self, session, project: str, name: str, tag: str = None, hash_key: str = None
+    ) -> schemas.FeatureSet:
+        query = self._query(session, FeatureSet, name=name, project=project)
+        computed_tag = tag or "latest"
+        tag_fs_uid = None
+        if not tag and hash_key:
+            uid = hash_key
+        else:
+            tag_fs_uid = self._resolve_tag_function_uid(
+                session, FeatureSet, project, name, computed_tag
+            )
+            if tag_fs_uid is None:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Feature-set tag not found {project}/{name}:{tag}"
+                )
+            uid = tag_fs_uid
+        if uid:
+            query = query.filter(FeatureSet.uid == uid)
+        obj = query.one_or_none()
+        if obj:
+            db_fs = self._transform_feature_set_model_to_scheme(obj)
+
+            # If connected to a tag add it to metadata
+            if tag_fs_uid:
+                db_fs.metadata.tag = computed_tag
+            return db_fs
+        else:
+            raise mlrun.errors.MLRunNotFoundError(
+                f"Feature-set not found {project}/{name}"
+            )
+
+    # TODO - IMPLEMENT QUERY BY LABEL!!!!!
+    def list_feature_sets(
+        self,
+        session,
+        project: str,
+        name: str = None,
+        tag: str = None,
+        state: str = None,
+        entities: List[str] = None,
+        features: List[str] = None,
+        labels: List[str] = None,
+    ) -> List[schemas.FeatureSet]:
+        # Find object IDs by tag, project and fs-name (which is a like query)
+        tag = tag or "latest"
+        tag_query = self._query(session, FeatureSet.Tag, project=project, name=tag)
+        if name:
+            tag_query = tag_query.filter(FeatureSet.Tag.obj_name.ilike(f"%{name}%"))
+        obj_uids = {row.obj_id: row.name for row in tag_query}
+
+        # Query exact fields
+        query = self._query(session, FeatureSet, project=project, state=state)
+
+        if name is not None:
+            query = query.filter(FeatureSet.name.ilike(f"%{name}%"))
+        if obj_uids:
+            query = query.filter(FeatureSet.id.in_(obj_uids.keys()))
+        if entities:
+            ent_alias = aliased(Feature)
+            query = query.join(FeatureSet.entities.of_type(ent_alias)).filter(
+                ent_alias.name.in_(entities)
+            )
```

Comment:
No. Alias is mandatory in this case. The problem is that this is potentially a double join on the same table (both features and entities point to same table - features). I must use alias in that case, otherwise the SQL query formed will not be valid - the condition will look for feature.name, but there are two feature tables in the join query.

Reply:
Ahhh, yes I see, very nice you handled this, please add some comment - something like "we must alias cause this is potentially a double join (when both features and entities are given)"

Reply:
Done.